pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaMPI.hpp

```diff
@@ -334,6 +334,39 @@ class StateVectorCudaMPI final
         mpi_manager_.Barrier();
     }
 
+    /**
+     * @brief Collapse the state vector after having measured one of the qubit.
+     *
+     * Note: The branch parameter imposes the measurement result on the given
+     * wire.
+     *
+     * @param wire Wire to measure.
+     * @param branch Branch 0 or 1.
+     */
+    void collapse(const std::size_t wire, const bool branch) {
+        PL_ABORT_IF_NOT(wire < this->getTotalNumQubits(),
+                        "Invalid wire index.");
+
+        const int wireInt =
+            static_cast<int>(this->getTotalNumQubits() - 1 - wire);
+
+        if (static_cast<std::size_t>(wireInt) < BaseType::getNumQubits()) {
+            // local wire
+            collapse_local_(wireInt, branch);
+        } else {
+            // global wire
+            int local_wire = 0;
+            std::vector<int2> wirePairs{make_int2(wireInt, local_wire)};
```

Comment:
```suggestion
            std::vector<int2> wirePairs{make_int2(wireInt, 0)};
```
I do not know if this is possible.

Reply:
let's keep the current implementation. `local_wire` is `constexpr` and we have to pass it the `MPI_Dispatcher` as well.